### PR TITLE
Fix IDE0017: Object initialization can be simplified

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [NA] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [A] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [NA] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [NA] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [NA] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 1]
```diff
@@ -273,17 +273,18 @@ protected override void ProcessRecord()
...
+                    Provider = _provider,
+                    Component = Component,
+                    Role = Role,
+                    Functionality = Functionality,
+                    ProviderContext = new ProviderContext(
                     this.Path,
                     this.Context.Engine.Context,
-                    this.SessionState.Path);
-                helpRequest.CommandOrigin = this.MyInvocation.CommandOrigin;
+                    this.SessionState.Path),
```
Comment (iSazonov) [inline src/System.Management.Automation/help/HelpCommands.cs:285]: Identations.

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 2]
```diff
@@ -826,12 +827,14 @@ public static string GetHelpUri(PSObject commandInfoPSObject)
...
-                    HelpRequest helpRequest = new HelpRequest(cmdName, cmdInfo.HelpCategory);
-                    helpRequest.ProviderContext = new ProviderContext(
+                    HelpRequest helpRequest = new HelpRequest(cmdName, cmdInfo.HelpCategory)
+                    {
+                        ProviderContext = new ProviderContext(
                         string.Empty,
                         currentContext,
-                        currentContext.SessionState.Path);
-                    helpRequest.CommandOrigin = CommandOrigin.Runspace;
+                        currentContext.SessionState.Path),
```
Comment (iSazonov) [inline src/System.Management.Automation/help/HelpCommands.cs:835]: The same.

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/help/HelpCommands.cs (hunk 1 above)
Comment (xtqqczze) [reply src/System.Management.Automation/help/HelpCommands.cs:285]: Seems there are bugs in IDE0017 code fix, I will open an issue in dotnet/roslyn

**Comment by daxian-dbw** [comment]
@PowerShell/powershell-maintainers reviewed this PR and agreed to reject it based on the following reason(s):
> The size of the code changes is large (+2,252, −1,449) and property initializer will be compiled to similar code (if not the same), so the chance of regression and confliction is high with relatively low value.

**Comment by iSazonov** [comment]
We already have `dotnet_style_object_initializer = true:suggestion` in EditorConfig.

Close based on PowerShell Maintainers conclusion.

**Comment by xtqqczze** [comment]
@daxian-dbw Could we reconsider this PR for 7.2 Preview1?

**Pull request closed without merging**